src/clj/puppetlabs/puppetserver/jruby_request.clj

```diff
@@ -57,18 +57,28 @@
 (defn wrap-with-environment-validation
   "Middleware function which validates the presence and syntactical content
   of an environment in a ring request.  If validation fails, a :bad-request
-  slingshot exception is thrown."
-  [handler]
-  (fn [request]
-    (let [environment (get-environment-from-request request)]
-      (cond
-        (nil? environment)
-        (ringutils/throw-bad-request!
-         (i18n/tru "An environment parameter must be specified"))
+  slingshot exception is thrown. If the optional-environment-param is
+  provided as true, it allows for the endpoint to have its query param
+  environment to be optional. However if the param is provided, it will
+  still validate that param for the given request."
+  ([handler optional-environment-param]
+   (fn [request]
+     (let [environment (get-environment-from-request request)]
+       (if (and optional-environment-param
```

Comment:
Hmm, yeah that looks like a problem to me too. I think your suggestion makes sense. I'll go head and update it to work that way.